pkg/dyninst/decode/marshal.go

```diff
@@ -201,441 +187,111 @@ func (d *Decoder) encodeValue(
-	kind, ok := irType.GetGoKind()
-	if !ok {
-		return fmt.Errorf("no go kind for type %s (ID: %d)", irType.GetName(), irType.GetID())
-	}
-	switch kind {
-	case reflect.Bool:
-		if len(data) < 1 {
-			return errors.New("passed data not long enough for bool")
-		}
-		return writeTokens(enc, jsontext.String(strconv.FormatBool(data[0] == 1)))
-	case reflect.Int:
-		if len(data) < 8 {
-			return errors.New("passed data not long enough for int")
-		}
-		return writeTokens(enc, jsontext.String(strconv.FormatInt(int64(binary.NativeEndian.Uint64(data)), 10)))
-	case reflect.Int8:
-		if len(data) < 1 {
-			return errors.New("passed data not long enough for int8")
-		}
-		return writeTokens(enc, jsontext.String(strconv.FormatInt(int64(int8(data[0])), 10)))
-	case reflect.Int16:
-		if len(data) < 2 {
-			return errors.New("passed data not long enough for int16")
-		}
-		return writeTokens(enc, jsontext.String(strconv.FormatInt(int64(binary.NativeEndian.Uint16(data)), 10)))
-	case reflect.Int32:
-		if len(data) != 4 {
-			return errors.New("passed data not long enough for int32")
-		}
-		return writeTokens(enc, jsontext.String(strconv.FormatInt(int64(binary.NativeEndian.Uint32(data)), 10)))
-	case reflect.Int64:
-		if len(data) != 8 {
-			return errors.New("passed data not long enough for int64")
-		}
-		return writeTokens(enc, jsontext.String(strconv.FormatInt(int64(binary.NativeEndian.Uint64(data)), 10)))
-	case reflect.Uint:
-		if len(data) != 8 {
-			return errors.New("passed data not long enough for uint")
-		}
-		return writeTokens(enc, jsontext.String(strconv.FormatUint(binary.NativeEndian.Uint64(data), 10)))
-	case reflect.Uint8:
-		if len(data) != 1 {
-			return errors.New("passed data not long enough for uint8")
-		}
-		return writeTokens(enc, jsontext.String(strconv.FormatUint(uint64(data[0]), 10)))
-	case reflect.Uint16:
-		if len(data) != 2 {
-			return errors.New("passed data not long enough for uint16")
-		}
-		return writeTokens(enc, jsontext.String(strconv.FormatUint(uint64(binary.NativeEndian.Uint16(data)), 10)))
-	case reflect.Uint32:
-		if len(data) != 4 {
-			return errors.New("passed data not long enough for uint32")
-		}
-		return writeTokens(enc, jsontext.String(strconv.FormatUint(uint64(binary.NativeEndian.Uint32(data)), 10)))
-	case reflect.Uint64:
-		if len(data) != 8 {
-			return errors.New("passed data not long enough for uint64")
-		}
-		return writeTokens(enc, jsontext.String(strconv.FormatUint(binary.NativeEndian.Uint64(data), 10)))
-	case reflect.Uintptr:
-		if len(data) != 8 {
-			return errors.New("passed data not long enough for uintptr")
+func (s *goSwissMapHeaderType) encodeSwissMapGroup(
+	d *Decoder,
+	enc *jsontext.Encoder,
+	groupData []byte,
+) (valuesEncoded int, err error) {
+	slotsData := groupData[s.slotsOffset : s.slotsOffset+s.slotsSize]
+	controlWord := binary.LittleEndian.Uint64(groupData[s.ctrlOffset : s.ctrlOffset+uint32(s.ctrlSize)])
+	entrySize := (s.valueFieldOffset - s.keyFieldOffset) + s.valueTypeSize
```

Comment:
this isn't right sadly. Consider a `bool` value. You'll run into the same problem. You need to record the slot size explicitly too. Add a test with bool values while you're at it and put a few entries in that map